feat: rex ia avec client part 2 (#107)
BearStudio/bearstudio.fr: src/content/posts/ia-et-client-les-vibecodeurs/en.md

---
title: 'IA & Client: Vibecoders'
slug: 'ia-and-client-vibecoders'
date: 2026-04-02
categories:
  - 'ia'
tags:
  - 'ia'
  - 'rex'
  - 'vibe-coding'
  - 'collaboration'
metaDescription: 'How we adapted our processes to collaborate with clients who vibe-code their own features with AI.'
excerpt: 'The client wants to vibe-code alongside us. First reaction: big sigh. But adapt or disappear. Here is how we rebuilt our production process around a new principle: we are no longer the authors of every line of code; we are the editors.'
thumbnail:
  image: 'images/thumbnail.jpg'
  alt: 'Illustration of a man working on a screen generated by AI next to a polar bear.'
authors:
  - 'heloise-guillaume'
---

## In the previous article

> In the first article, I presented how our clients now use AI to prototype their ideas, turning these "shadow prototypes" into real specifications.

Reviewer: ```suggestion
In the first article about client "Prototypers", I presented how our clients now use AI to prototype their ideas, turning these "shadow prototypes" into real specifications.
```
\+ _In the first article about client "Prototypers"_ should be a link redirecting to the article

> But a new chapter begins when clients are no longer content with just showing... they want to **build with us**.

Reviewer: ```suggestion
But a new chapter begins when clients are no longer content to simply exhibit... they want to **build with us**.
```

## Surviving the Rise of Vibe Coders Clients: How We Adapt

> Client A's case was the first step: accepting AI as a specification tool. But with another client (let's call them "Team B") we faced a much more chaotic reality.

Reviewer: ```suggestion
Client A's case was the first step: accepting AI as a specification tool. But with another client (let's call them "Client B") we faced a much more chaotic reality.
```
It makes more sense to keep a variant of the same name
\+ To be reflected throughout the article

The problem? They did not tell us at first. One day, they simply sent me a pull request fully vibe-coded with no context, no ticket description, and nothing even listed in the backlog.

Our first reaction was confusion. I spent hours reviewing the code, trying to reverse-engineer the intent, the impact, and what needed to be tested. It was unsustainable. I couldn't just "fix" their code; we had to fix the collaboration.

Finally, they came to us with a proposal that would give most traditional developers a heart attack:

> We don't have the budget for you to build everything. We want to vibe-code features alongside you. You handle the core; we'll handle the rest.

The old reflex:

![annoyed](images/annoyed.jpg)

Two years ago, our answer would have strongly discouraged this because of the review and maintenance cost.

In the traditional agency model, the development team retains full responsibility for the production codebase. Contributions are therefore tightly controlled to ensure consistency, reliability, and security. Any external changes can introduce defects or risks that the team remains accountable for, making unmanaged contributions difficult to sustain.

> But as Rudy often says: **"Adapt or disappear."** If we refused, Team B would have just hired a freelancer who didn't care about quality, or they would have tried to do it all themselves and crashed six months later.

Reviewer: Maybe add a link on "Rudy" to his team member page

## The new process: the "guardrails" strategy

We accepted the challenge. But to make it work, we had to tear down our production process and rebuild it around a new concept: **We are no longer the authors of every line of code; we are the editors.**

Here is the exact workflow we implemented to survive "vibe coding":

### 1. The "pre-code" check (the planning phase)

Team B uses Claude Code with the "Planning" feature. Now, instead of coding immediately, they generate the Plan. We intervene here. We review their human brief and the AI's proposed execution plan. We iterate on the logic before a single line of code is generated. If the plan is flawed, the code will be garbage. We fix the blueprint, not the house, by iterating until we are both satisfied: we simplify the architecture, optimize processes, verify the different business cases upfront...

> ### 2. Context is king (agents.md)

Reviewer: ```suggestion
### 2. Context is key (agents.md)
```
